DeepLinkKit/Router/DPLDeepLinkRouter.h

```diff
@@ -67,7 +67,10 @@ typedef void(^DPLRouteCompletionBlock)(BOOL handled, NSError *error);
  */
 - (void)registerBlock:(DPLRouteHandlerBlock)routeHandlerBlock forRoute:(NSString *)route;
 
-
+/**
+ Returns a Deep link for the given URL, if one exists. Returns `nil` if no matching routes are found for the URL.
+ */
+- (nullable DPLDeepLink *)deepLinkForURL:(nonnull NSURL *)url;
```

Comment:
This method should sit at the core of route matching if it exists, so definitely needs tests 😄